podium/podium-charts
# Add autobuild precommit script

## Brief context on code (tell us why these changes are necessary)
Adds a pre-commit hook script that will automatically run the build for all of your staged changes. No more needing to `yarn build` 😄  

Now you can go from this:

![](https://media.giphy.com/media/l0CRCPXURKX7lyU48/giphy.gif)

To this:

![](https://media.giphy.com/media/dkGhBWE3SyzXW/giphy.gif)

## Test plan

There are a few things you should test with this as it should handle a few use cases. It should **not** build any files of uncommitted or unstaged changes. Here are a few scenarios: 

1. Make a change that is build-able (stories work well for this). Stage and commit the change. You should see the build run for that particular file. 

2. Make 2 changes, and only stage one of them. Commit and then verify that only the file you staged has an updated build file.


** Test Cases **

1: Only staged local changes, all have build files
![image](https://user-images.githubusercontent.com/4327477/55755249-c3586100-5a0b-11e9-8553-4775e7dbe730.png)

2. Staged changes have build files, unstaged changes don't
![image](https://user-images.githubusercontent.com/4327477/55755346-fdc1fe00-5a0b-11e9-88c9-0b2fa4310ef3.png)

3. Staged new file (Test.story.jsx) includes build files
![image](https://user-images.githubusercontent.com/4327477/55755555-850f7180-5a0c-11e9-8308-69d3323417ce.png)

4. Unstaged new file (Test3.story.jsx) does not have build files
![image](https://user-images.githubusercontent.com/4327477/55755948-a91f8280-5a0d-11e9-901d-78e65035f1f3.png)

5.  Deleting a file also removes the build files
Starting with these files:
![image](https://user-images.githubusercontent.com/4327477/55811130-e20b3600-5aa5-11e9-89d5-b23b3801280b.png)

You should then end up with this: 
![image](https://user-images.githubusercontent.com/4327477/55811242-1121a780-5aa6-11e9-812d-c5a5ff6f2793.png)

6. Process exits when there is a failed build
With a syntax error or anything that will fail the build, the expected output runs:
![image](https://user-images.githubusercontent.com/4327477/55812061-7c1fae00-5aa7-11e9-8d19-0a15999f4245.png)


## Before asking for code review

- [x] I have unit tested behavioral changes
- [x] I have verified test plan works
- [x] I have merged the base branch into this branch
- [x] I have ensured that CI is passing
- [x] I have filled out the "Brief context on code" and "Test plan" sections above
- [x] I have ~rerun the build and~ updated the version in package.json
